nixos/ec2/autohostname.nix

```diff
@@ -44,7 +45,7 @@ let
 
     signature=$(echo -n $date | ${openssl} dgst -binary -sha1 -hmac $1 | ${base64})
     auth_header="X-Amzn-Authorization: AWS3-HTTPS AWSAccessKeyId=$2,Algorithm=HmacSHA1,Signature=$signature"
-    hostname=$(${hostname}).${zone}
+    hostname=$(${hostname}); ${fgrep} -q ${zone} <<< $hostname || hostname=$hostname.${zone}
```

Comment:
I mean here: https://github.com/zalora/microgram/blob/fqdn-lol/nixos/ec2/autohostname.nix#L94

@ip1981 why not?